components/infobox/extensions/commons/placement_stats.lua

```diff
@@ -0,0 +1,133 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:InfoboxPlacementStats
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local placementStats = {}
+local Class = require('Module:Class')
+local Template = require('Module:Template')
+local Table = require('Module:Table')
+local Medal = require('Module:Medal')
+local Settings = require('Module:InfoboxPlacementStats/Settings')
+
+local _PLACEMENTS = { '1', '2', '3', '>' }
+
+function placementStats.get(args)
+	args = args or {}
+	local mode = args.mode or ''
+	local participant = args.participant or mw.title.getCurrentTitle().prefixedText
+
+	local queryData = placementStats._getQueryData(mode, participant)
+
+	return placementStats:_makeTable(queryData)
+end
+
+function placementStats._getQueryData(mode, participant)
+	local sortedData = { total = { top3 = 0, total = 0 } }
+	for _, place in pairs(_PLACEMENTS) do
+		sortedData.total[place] = 0
+	end
+
+	local query = 'count::placement'
+	local baseConditions = '[[mode::' .. mode .. ']] AND [[participant::' .. participant .. ']]'
+
+	if not Table.isEmpty(Settings.exclusionTypes) then
+		baseConditions = baseConditions .. ' AND [[liquipediatiertype::!'
+			.. table.concat(Settings.exclusionTypes, ']] AND [[liquipediatiertype::!')
+			.. ']]'
+	end
+
+	for _, tier in pairs(Settings.tiers) do
+		sortedData[tier] = { top3 = 0 }
+		for _, place in pairs(_PLACEMENTS) do
+			local queryData = mw.ext.LiquipediaDB.lpdb('placement', {
+				conditions = baseConditions ..
+					' AND [[liquipediatier::' .. tier .. ']] AND [[placement::' .. place .. ']]',
+				query = query,
+			})
+			local count = tonumber(queryData[1].count_placement or 0) or 0
```

Comment:
Can `queryData` ever be length 0?

Reply:
since we query for `count::placement` it should always return a value
(if no such placements exist it should return 0 afaik)

Reply:
Okay.